.github/workflows/package.yaml

```diff
@@ -89,16 +89,33 @@ jobs:
         run: nix develop -c just package ${{ inputs.target }}
 
       - name: Test Package debian
+        if: vars.TEST_OS_PACKAGE_ENABLED == 'true'
         run: |
+          cat ${{ secrets.TEST_OS_PACKAGE_HOPR_IDENTITY_FILE }} > ${{ github.workspace }}/deploy/nfpm/hopr.id
+          chmod 600 ${{ github.workspace }}/deploy/nfpm/hopr.id
           nix develop -c just test-package deb ${{ inputs.target }}
+        env:
+          HOPRD_PASSWORD: ${{ secrets.TEST_OS_PACKAGE_HOPRD_PASSWORD }}
+          HOPRD_SAFE_ADDRESS: ${{ secrets.TEST_OS_PACKAGE_HOPRD_SAFE_ADDRESS }}
+          HOPRD_MODULE_ADDRESS: ${{ secrets.TEST_OS_PACKAGE_HOPRD_MODULE_ADDRESS }}
+          HOPRD_PROVIDER: ${{ secrets.TEST_OS_PACKAGE_HOPRD_PROVIDER }}
 
       - name: Test Package centos
-        run: |
-          nix develop -c just test-package rpm ${{ inputs.target }}
+        if: vars.TEST_OS_PACKAGE_ENABLED == 'true'
+        run: nix develop -c just test-package rpm ${{ inputs.target }}
+        env:
+          HOPRD_PASSWORD: ${{ secrets.TEST_OS_PACKAGE_HOPRD_PASSWORD }}
+          HOPRD_SAFE_ADDRESS: ${{ secrets.TEST_OS_PACKAGE_HOPRD_SAFE_ADDRESS }}
+          HOPRD_MODULE_ADDRESS: ${{ secrets.TEST_OS_PACKAGE_HOPRD_MODULE_ADDRESS }}
+          HOPRD_PROVIDER: ${{ secrets.TEST_OS_PACKAGE_HOPRD_PROVIDER }}
 
       - name: Test Package archlinux
-        run: |
-          nix develop -c just test-package archlinux ${{ inputs.target }}
+        run: nix develop -c just test-package archlinux ${{ inputs.target }}
+        env:
+          HOPRD_PASSWORD: ${{ secrets.TEST_OS_PACKAGE_HOPRD_PASSWORD }}
+          HOPRD_SAFE_ADDRESS: ${{ secrets.TEST_OS_PACKAGE_HOPRD_SAFE_ADDRESS }}
+          HOPRD_MODULE_ADDRESS: ${{ secrets.TEST_OS_PACKAGE_HOPRD_MODULE_ADDRESS }}
+          HOPRD_PROVIDER: ${{ secrets.TEST_OS_PACKAGE_HOPRD_PROVIDER }}
```

Comment:
Should this not be a matrix?

Reply:
We could be we would not take advantage of the parallelization. Using a matrix would involve to create a separate workflow run (let's call it test-package) only for that, and then in this job (package), split it and once the matrix test-package job finishes publish the pacakges in a third job(let's call it publish-package).
As the last step of publishing the packages in the workflow run would be a separate job, I would have to manage upload/download artifacts again.